Firestore/core/src/firebase/firestore/api/handle_maybe.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_API_HANDLE_MAYBE_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_API_HANDLE_MAYBE_H_
+
+#include <functional>
+
+#include "Firestore/core/src/firebase/firestore/util/statusor.h"
+
+namespace firebase {
+namespace firestore {
+namespace api {
+
+template <typename T>
+using HandleMaybe = std::function<void(util::StatusOr<T>)>;
```

Comment:
It seems to me like a callback that receives either a result or an error is common enough to warrant a helper type. That said:
* I'm very open to suggestions regarding the name (by the way, similarly to `StatusOr`, the type is supposed to be a part of the name, so that it turns into something like "handle maybe DocumentSnapshot");
* perhaps it should be in `util` namespace, not `api`?

Reply:
Since this is a callback that accepts a `StatusOr`, why not `StatusOrCallback`? If you accept that name, it seems as if should be in `util`. Or maybe as a member in StatusOr?

However, it's not really saving us all that much, and codesearching in google3, I see a lot of things accepting `std::function<void(StatusOr<T>)>` directly.